SimCalorimetry/EcalEBTrigPrimProducers/plugins/EcalEBTrigPrimPhase2Producer.cc

```diff
@@ -0,0 +1,185 @@
+  //
+  edm::ESHandle<EcalEBPhase2TPGTimeWeightIdMap> theEcalEBTPGTimeWeightIdMap_handle =
+      setup.getHandle(theEcalEBTPGTimeWeightIdMap_Token_);
+  const EcalEBPhase2TPGTimeWeightIdMap* ecaltpgTimeWeightMap = theEcalEBTPGTimeWeightIdMap_handle.product();
+  //
+  edm::ESHandle<EcalTPGWeightGroup> theEcalTPGWeightGroup_handle = setup.getHandle(theEcalTPGWeightGroup_Token_);
+  const EcalTPGWeightGroup* ecaltpgWeightGroup = theEcalTPGWeightGroup_handle.product();
+  //
+  //edm::ESHandle<EcalTPGTowerStatus> theEcalTPGTowerStatus_handle = setup.getHandle(theEcalTPGTowerStatus_Token_);
+  //const EcalTPGTowerStatus* ecaltpgBadTT = theEcalTPGTowerStatus_handle.product();
+  //
+  //edm::ESHandle<EcalTPGSpike> theEcalTPGSpike_handle = setup.getHandle(theEcalTPGSpike_Token_);
+  //const EcalTPGSpike* ecaltpgSpike = theEcalTPGSpike_handle.product();
+
+  ////////////////
+  algo_->setPointers(ecaltpPed, ecaltpLin, ecaltpgBadX, ecaltpgAmplWeightMap, ecaltpgTimeWeightMap, ecaltpgWeightGroup);
+
+  return setup.get<EcalEBPhase2TPGLinearizationConstRcd>().cacheIdentifier();
+  //  return setup.get<EcalLiteDTUPedestalsRcd>().cacheIdentifier();
+}
+
+void EcalEBTrigPrimPhase2Producer::endRun(edm::Run const& run, edm::EventSetup const& setup) { algo_.reset(); }
+
+EcalEBTrigPrimPhase2Producer::~EcalEBTrigPrimPhase2Producer() {}
+
+// ------------ method called to produce the data  ------------
+void EcalEBTrigPrimPhase2Producer::produce(edm::Event& e, const edm::EventSetup& iSetup) {
+  nEvent_++;
+
+  // get input collections
+  edm::Handle<EBDigiCollectionPh2> barrelDigiHandle;
+
+  if (!e.getByToken(tokenEBdigi_, barrelDigiHandle)) {
+    edm::EDConsumerBase::Labels labels;
+    labelsForToken(tokenEBdigi_, labels);
+    edm::LogWarning("EcalTPG") << " Couldnt find Barrel digis " << labels.module << " and label "
+                               << labels.productInstance << "!!!";
+  }
+
+  if (debug_)
+    std::cout << "EcalTPG"
+              << " =================> Treating event  " << nEvent_ << ", Number of EB digis "
+              << barrelDigiHandle.product()->size() << std::endl;
+
+  auto pOut = std::make_unique<EcalEBPhase2TrigPrimDigiCollection>();
+  auto pOutTcp = std::make_unique<EcalEBPhase2TrigPrimDigiCollection>();
+
+  //std::cout << " Event number " << e.id().event() << std::endl;
+
+  // invoke algorithm
+
+  const EBDigiCollectionPh2* ebdigi = nullptr;
+  ebdigi = barrelDigiHandle.product();
+  algo_->run(ebdigi, *pOut);
+
+  if (debug_) {
+    std::cout << "produce"
+              << " For Barrel  " << pOut->size() << " TP  Digis were produced" << std::endl;
+  }
+
+  std::cout << "produce"
+            << " For Barrel  " << pOut->size() << " TP  Digis were produced" << std::endl;
+
+  //  debug prints if TP >0
+  int nonZeroTP = 0;
+
+  if (debug_)
+    std::cout << "EcalTPG Printing only non zero TP " << std::endl;
+
+  int nXstal = 0;
+  for (unsigned int i = 0; i < pOut->size(); ++i) {
+    nXstal++;
+
+    if (debug_) {
```

Comment:
Consider moving this `if` outside of the `for` loop to not loop for nothing if `debug_` is `false`.

Reply:
No, sorry. I need to update the counter nXstals. Not for production ... for my tests and I do not like to re-write something every time I need to test/debug. For the time being I leave all as it is

Reply:
If `nXtals` is needed even when debugging is off you could still to it like this
```c++
if (debug_) {
  for (unsigned int i = 0; i < pOut->size(); ++i) {
    ++nXtals;
    // rest of the debug code
  }
} else {
  nXtals += pOut->size();
}
```
Even if `nXtals` seems to be only used inside the for loop at the moment.

Reply:
Unless your tests add additional code that is not in this PR I think the suggestion above would have the same effect without the need for looping when not in debug mode.